### ISSUE TITLE
[a11y]: None of the descendent elements with "menuitem" role is tabbable - OverflowMenu on v10

### ISSUE THREAD

**Issue by kenneth-nguyen1** [carbon-design-system/carbon#14602]
### Package

carbon-components, carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

carbon-components: "10.58.8", carbon-components-react: "7.59.10"

### React version

v17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment (v3.1.59)

### Description

I saw that this accessibility violation was fixed for Carbon v11 in this issue https://github.com/carbon-design-system/carbon/issues/12728, but wanted to request if it can also get fixed for v10.

This is still an issue on the latest Carbon v10 storybook running on package version 7.59.13
<img width="1763" alt="carbon-v10-overflow-avt-issue" src="https://github.com/carbon-design-system/carbon/assets/114688923/1ec6ea84-0a6d-4ed1-af05-0d8f7b6f7e83">

### Reproduction/example

https://v7-react.carbondesignsystem.com/?path=/story/components-overflowmenu--default

### Steps to reproduce

1. Go to Carbon v10 storybook https://v7-react.carbondesignsystem.com/?path=/story/components-overflowmenu--default
2. Navigate to the OverflowMenu component story
3. Click the overflow menu icon to show the dropdown menu
4. Run the IBM Equal Access Accessibility Checker scan
5. See the specified violation `None of the descendent elements with "menuitem" role is tabbable`

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
Fixed via #14606, and included in [v10.59.14](https://github.com/carbon-design-system/carbon/releases/tag/v10.59.14)
